### PR TITLE
Odfe it framework release

### PR DESCRIPTION
**Pull request by sidheart**
*Issue #, if available:* 163, 164, 166, 22

*Description of changes:* Adds a basic framework for executing integration tests against ElasticSearch, a test for this framework, and build logic to ensure that the test suite executes on every check in.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -40,6 +39,10 @@ jobs:
...
       working-directory: ./tmp/pa
       run: ./gradlew updateShas
+    - name: Set docker-compose path
+      run: echo ::set-env name=DOCKER_COMPOSE_LOCATION::$(which docker-compose)
+    - name: Set vm.max_map_count
+      run: sudo sysctl -w vm.max_map_count=262144
     - name: Start Build
       working-directory: ./tmp/pa
-      run: ./gradlew build
+      run: ./gradlew build --info --stacktrace
```
Comment (yojs) [inline .github/workflows/gradle.yml:48]: The info creates a lot of details and the actual errors are somewhere up. Do we really want this ?

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerIT.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    @Test
+    public void checkMetrics() throws Exception {
+        ensurePaAndRcaEnabled();
+        Request request = new Request("GET",
+                "/_opendistro/_performanceanalyzer/metrics/?metrics=Disk_Utilization&agg=max&dim=&nodes=all");
+        Response resp = paClient.performRequest(request);
+        assert resp.getStatusLine().getStatusCode() == HttpStatus.SC_OK;
+        LOG.info("PA is emitting metrics!! {}", EntityUtils.toString(resp.getEntity(), "UTF-8"));
+        System.out.println("WASSUP");
```
Comment (yojs) [inline src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerIT.java:71]: Remove ?

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerIT.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    @Test
+    public void checkMetrics() throws Exception {
+        ensurePaAndRcaEnabled();
+        Request request = new Request("GET",
+                "/_opendistro/_performanceanalyzer/metrics/?metrics=Disk_Utilization&agg=max&dim=&nodes=all");
+        Response resp = paClient.performRequest(request);
+        assert resp.getStatusLine().getStatusCode() == HttpStatus.SC_OK;
+        LOG.info("PA is emitting metrics!! {}", EntityUtils.toString(resp.getEntity(), "UTF-8"));
```
Comment (yojs) [inline src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerIT.java:70]: In addition we should also check that the json response has all the expected keys with non-negative values

File: .github/workflows/gradle.yml [hunk 4]
```diff
@@ -40,6 +39,10 @@ jobs:
     - name: Update SHA
       working-directory: ./tmp/pa
       run: ./gradlew updateShas
+    - name: Set docker-compose path
+      run: echo ::set-env name=DOCKER_COMPOSE_LOCATION::$(which docker-compose)
+    - name: Set vm.max_map_count
```
[the remark this answers is not in the document]
Comment (sidheart) [reply .github/workflows/gradle.yml:47]: Done

File: .github/workflows/gradle.yml (hunk 1 above)
Comment (sidheart) [reply .github/workflows/gradle.yml]: Removed

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerIT.java (hunk 2 above)
Comment (sidheart) [reply src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerIT.java]: I was very excited the first time it worked, removed

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerIT.java (hunk 3 above)
Comment (sidheart) [reply src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerIT.java]: Done

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerIT.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+        // Field related strings
+        public static final String FIELDS = "fields";
+        public static final String FIELD_NAME = "name";
+        public static final String FIELD_TYPE = "type";
+        public static final String DOUBLE_TYPE = "DOUBLE";
+
+        // Metrics related strings
+        public static final String M_DISKUTIL = "Disk_Utilization";
+    }
```
Comment (ktkrg) [inline src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerIT.java:77]: nit: move the class definition to the end of the class.

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerIT.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+    @Test
+    public void checkMetrics() throws Exception {
+        ensurePaAndRcaEnabled();
+        Request request = new Request("GET",
+                "/_opendistro/_performanceanalyzer/metrics/?metrics=Disk_Utilization&agg=max&dim=&nodes=all");
+        Response resp = paClient.performRequest(request);
+        Assert.assertEquals(HttpStatus.SC_OK, resp.getStatusLine().getStatusCode());
+        ObjectMapper mapper = new ObjectMapper();
```
Comment (ktkrg) [inline src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerIT.java:86]: Do we need a new ObjectMapper here? Can we re-use the one defined on line 28?

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerIT.java (hunk 5 above)
Comment (sidheart) [reply src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerIT.java]: Done

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerIT.java (hunk 6 above)
Comment (sidheart) [reply src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerIT.java]: Fixed

**Pull request merged**